fastlane/swift/MainProcess.swift

```diff
@@ -47,9 +47,9 @@ class MainProcess {
             let PATH = run("/bin/bash", "-c", "-l", "eval $(/usr/libexec/path_helper -s) ; echo $PATH").stdout
             main.env["PATH"] = PATH
             let path = main.run(bash: "which fastlane").stdout
-            let pids = main.run("lsof", "-t", "-i", ":2000").stdout.split(separator: "\n")
+            let pids = main.run("lsof", "-t", "-i", ":\(argumentProcessor.port)").stdout.split(separator: "\n")
             pids.forEach { main.run("kill", "-9", $0) }
-            rubySocketCommand = main.runAsync(path, "socket_server", "-c", "1200")
+            rubySocketCommand = main.runAsync(path, "socket_server", "-c", "1200", "-p", argumentProcessor.port)
```

Comment:
The timeout is also taken in the `ArgumentProcessor` if you want it to be included as a parameter as well:
```suggestion
            rubySocketCommand = main.runAsync(path, "socket_server", "-c", argumentProcessor.timeout, "-p", argumentProcessor.port)
```